semantic-typescript/src/Language/TypeScript/Tags.hs

```diff
@@ -24,19 +30,19 @@ import Source.Source as Source
 import Tags.Tag
 import qualified Tags.Tagging.Precise as Tags
 
-class ToTags t where
+class ToTags (t :: ((* -> *) -> * -> *)) where
   tags ::
     ( Has (Reader Source) sig m,
       Has (Writer Tags.Tags) sig m
     ) =>
-    t Loc ->
+    t Parse.Err Loc ->
     m ()
   default tags ::
     ( Has (Reader Source) sig m,
       Has (Writer Tags.Tags) sig m,
-      Traversable1 ToTags t
+      Traversable1 ToTags (t Parse.Err)
```

Comment:
I added `Parse.Err` to this constraint in response to a directive issued by a prior error message:

```
    • Expecting one more argument to ‘Traversable1 (ToTags t)’
      Expected a constraint,
        but ‘Traversable1 (ToTags t)’ has kind ‘(* -> *) -> Constraint’
    • In the type signature:
        tags :: (Has (Reader Source) sig m,
                 Has (Writer Tags.Tags) sig m,
                 Traversable1 (ToTags t)) =>
                t Parse.Err Loc -> m ()
      In the class declaration for ‘ToTags’
   |
43 |       Traversable1 (ToTags t)
   |       ^^^^^^^^^^^^^^^^^^^^^^^
```

However,  this addition only results in the following error:

```
Tags.hs:43:20-25: error:
    • Expected kind ‘(* -> *) -> Constraint’,
        but ‘ToTags’ has kind ‘((* -> *) -> * -> *) -> Constraint’
    • In the first argument of ‘Traversable1’, namely ‘ToTags’
      In the type signature:
        tags :: (Has (Reader Source) sig m,
                 Has (Writer Tags.Tags) sig m,
                 Traversable1 ToTags (t Parse.Err)) =>
                t Parse.Err Loc -> m ()
      In the class declaration for ‘ToTags’
   |
43 |       Traversable1 ToTags (t Parse.Err)
   | 
```

I hypothesized that the `ConstraintKinds` extension could be used to treat constraints as first-class citizens, but I think I'm conflating a few too many things about constraint kinds. @patrickt could use some 🍐-ing help here.